ONSdigital/eq-author-app
# Permissions - Questionnaire editors

### What is the context of this PR?
This adds the ability to add editors via the sharing modal. By default a user only has read permissions unless they are the creator.

### How to review 
1. Ensure that you can't update a questionnaire when in read only mode
2. Ensure that you can update a questionnaire when either a creator or editor

### Interesting things
1. We weren't clearing the apollo cache on logout so the permissions were persisted in the UI view but this happens now.
2. API tests were often crashing in watch mode as it was attempting to trace it and we didn't have a tracing server set up. I've now disabled tracing in test mode stubbed this dependency when we do need it enabled
3. All mutation resolvers are now all wrapped in a function that adds the saving and returns an error if the user does not have permission
